clusterdock/topologies/cdh/actions.py

```diff
@@ -215,3 +281,110 @@ def remove_files(cluster, files, nodes):
+    krb5_conf_data = config_files[KDC_KRB5_CONF_FILENAME]
+    krb5_conf_data = krb5_conf_data.replace('EXAMPLE.COM', network.upper())
+    krb5_conf_data = krb5_conf_data.replace('kerberos.example.com', '{}.{}'.format(KDC_HOST_NAME, network))
+    krb5_conf_data = krb5_conf_data.replace('example.com', network)
+    config_files[KDC_KRB5_CONF_FILENAME] = krb5_conf_data
+
+    kdc_conf_data = config_files[KDC_CONF_FILENAME]
+    kdc_conf_data = kdc_conf_data.replace('EXAMPLE.COM', network.upper())
+    conf_list = kdc_conf_data.split('\n')
+    index = conf_list.index('[kdcdefaults]')
+    conf_list.insert(index + 1, ' max_renewablelife = 7d')
+    conf_list.insert(index + 1, ' max_life = 1d')
+    kdc_conf_data = '\n'.join(conf_list)
+    config_files[KDC_CONF_FILENAME] = kdc_conf_data
+
+    acl_data = config_files[KDC_ACL_FILENAME]
+    acl_data = acl_data.replace('EXAMPLE.COM', network.upper())
+    config_files[KDC_ACL_FILENAME] = acl_data
+
+    # Serialize them back to the kdc Docker container
+    for kdc_conf_file in KDC_CONFIG_FILES:
+        o_tarstream = io.BytesIO()
+        with tarfile.open(fileobj=o_tarstream, mode='w') as tarfile_:
+            encoded_file_contents = config_files[kdc_conf_file].encode()
+            tarinfo = tarfile.TarInfo(basename(kdc_conf_file))
+            tarinfo.size = len(encoded_file_contents)
+            tarfile_.addfile(tarinfo, io.BytesIO(encoded_file_contents))
+        o_tarstream.seek(0)
+        client.put_archive(container=kdc_node.container_id, path=dirname(kdc_conf_file), data=o_tarstream)
+
+def _setup_kdc_server(cluster, nodes, network, kerberos_principals):
+    logger.info('Setting up kdc server ...')
+    principal_cmds = ""
+    if kerberos_principals:
+        princ_list = kerberos_principals.split(',')
+        create_principals = " ".join(["kadmin.local -q \"addprinc -randkey %s@%s\"; " % (item, network)
+                                      for item in princ_list])
+        create_keytab = "kadmin.local -q \"xst -norandkey -k %s %s\"; " % (KDC_KEYTAB_FILENAME,
+                                                                           ' '.join(['%s@%s' % (item, network) for item in princ_list]))
+        principal_cmds = '{} {}'.format(create_principals, create_keytab)
+
+    kdc_commands = ("kdb5_util create -s -r {realm} -P kdcadmin; "
+                    "kadmin.local -q \"addprinc -pw {adminpw} admin/admin@{realm}\"; "
+                    "kadmin.local -q \"addprinc -pw {cldradminpw} {cldradmin}@{realm}\"; "
+                    "{create_principals_keytab} krb5kdc; "
+                    "kadmind; "
+                    "authconfig --enablekrb5 --update; "
+                    "service sshd restart; "
+                    "service krb5kdc restart; "
+                    "service kadmin restart"
+                   ).format(realm=network,
+                            adminpw='acladmin',
+                            create_principals_keytab=principal_cmds,
```

Comment:
This is a bit confusing to read. The variable name is `create_principals_keytab` which makes me think it'll get a value of a keytab in some way, but then commands are passed in. Perhaps rename the variable to reflect that it's a command?